pyvrp/minimise_fleet.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+
+from pyvrp._pyvrp import ProblemData, VehicleType
+from pyvrp.solve import SolveParams, solve
+from pyvrp.stop import FirstFeasible, MultipleCriteria, StoppingCriterion
+
+
+def minimise_fleet(
+    data: ProblemData,
+    stop: StoppingCriterion,
+    seed: int = 0,
+    params: SolveParams = SolveParams(),
+) -> VehicleType:
+    """
+    Attempts to reduce the number of vehicles needed to achieve a feasible
+    solution to the given problem instance, subject to a stopping criterion.
+
+    .. warning::
+
+       This function is currently unable to solve instances with multiple
+       vehicle types. Support for such a setting may be added in future
+       versions of PyVRP.
+
+    Parameters
+    ----------
+    data
+        Problem instance with a given vehicle composition.
+    stop
+        Stopping criterion that determines how much effort to spend on finding
+        smaller fleet compositions.
+    seed
+        Seed value to use for the random number stream. Default 0.
+    params
+        Solver parameters to use. If not provided, a default will be used.
+
+    Returns
+    -------
+    VehicleType
+        The smallest fleet composition admitting a feasible solution to the
+        problem instance that could be found before the stopping criterion was
+        hit. The original fleet is returned if no feasible solution was found.
+
+    Raises
+    ------
+    ValueError
+        When the instance contains more than one vehicle type. That setting is
+        not yet supported.
+    """
+    if data.num_vehicle_types != 1:
+        msg = "Fleet minimisation does not understand multiple vehicle types."
+        raise ValueError(msg)
```

Comment:
I want to generalise this at some point, but it's hard to do well and would take research time that I do not currently have. Could be cool for a master's thesis (I'll keep that in mind when I start in Rotterdam in September).